File: pythonStudy/BOJ/1197 최소 스패닝 트리.py
# 1922 네트워크 연결
input = __import__('sys').stdin.readline
__import__('sys').setrecursionlimit(10**9)

v = int(input())
e = int(input())# 1197 최소 스패닝 트리
input = __import__('sys').stdin.readline
__import__('sys').setrecursionlimit(10**9)

v, e = map(int, input().split())
edge = [tuple(map(int, input().split())) for _ in range(e)]
grpNo = [i for i in range(v+1)]

def find(v):
    if v == grpNo[v]: return v
    grpNo[v] = find(grpNo[v])
    return grpNo[v]
def unite(v1, v2):
    if v2<v1: v1, v2 = v2, v1
    grpNo[find(v2)] = find(v1)

edge.sort(key = lambda x:x[2])
tCost = 0
for ne in edge:
    a, b, c = ne
    if find(a) != find(b):
        unite(a, b)
        tCost+=c
print(tCost)
edge = []
for i in range(e):
    a, b, c = map(int, input().split())
    if a!=b: edge.append((a, b, c))
grpNo = [i for i in range(v+1)]

def find(v):
    if v == grpNo[v]: return v
    grpNo[v] = find(grpNo[v])
    return grpNo[v]
def unite(v1, v2):
    if v2<v1: v1, v2 = v2, v1
    grpNo[find(v2)] = find(v1)

edge.sort(key = lambda x:x[2])
tCost = 0
for ne in edge:
    a, b, c = ne
    if find(a) != find(b):
        unite(a, b)
        tCost+=c
print(tCost)

File: cpp/Solved/BOJ/1197 최소 스패닝 트리.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
// 1197 최소 스패닝 트리
#define FOR(i, n, m) for(ll i=(n); i<(m); i++)

vector<int> adj[10002];
int grpNo[10002];
vector<tuple<int, int, int>> edge;
int find(int v) {
	if (grpNo[v] == v) return v;
	grpNo[v] = find(grpNo[v]);
	return grpNo[v];
}
void unite(int v1, int v2) {
	if (v2 < v1) swap(v1, v2);
	grpNo[find(v2)] = find(v1);
}
int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);

	int v, e;
	cin >> v >> e;
	FOR(i, 0, e) {
		int a, b, c;
		cin >> a >> b >> c;
		edge.push_back(tie(c, a, b));
	}
	FOR(i, 0, v) grpNo[i] = i;
	sort(edge.begin(), edge.end());
	long long TotalCost = 0;
	for (auto ne : edge) {
		int cost = get<0>(ne), a = get<1>(ne), b = get<2>(ne);
		if (find(a) != find(b)) {
			TotalCost += cost;
			unite(a, b);
		}
	}
	cout << TotalCost;
}